src/main/scala/it/pagopa/interop/partyprocess/service/impl/PartyProcessMailer.scala

```diff
@@ -26,6 +31,12 @@ trait PartyProcessMailer extends MailEngine { interopMailer: DefaultInteropMaile
     )
 
     interopMailer.sendWithTemplate(mailData)
+      .map(_ => logger.info(s"[$emailPurpose] Email successful sent"))
+      .recover(t => {
+        logger.error(s"[$emailPurpose] An error occurred while sending email", t)
```

Comment:
@iwoak is it ok the following pattern to log email sent result:

- [onboarding-contract-email] Email successful sent

- [onboarding-contract-email] An error occurred while sending email

Reply:
in this scope I have not the http status...
@beetlecrunch  do you know if it's possible to retrieve it?

Reply:
The `DefaultInteropMailer` uses [Courier](https://github.com/dmurvihill/courier), and it seem at a first glance that the error is modeled using a failing Future (i.e. an exception). The errors the library throws are just a few, apart for the ones thrown [here](https://github.com/dmurvihill/courier/blob/master/courier/src/main/scala/mailer.scala#L37=) by `javax.mail.Transport`. I don't think it's possible to extract from there any status code.

Reply:
OTOH, If you want to inject a specific status code in case the mail fails to deliver you have to take a look [here](https://github.com/pagopa/interop-be-party-process/blob/0.1.x/src/main/scala/it/pagopa/interop/partyprocess/api/impl/ProcessApiServiceImpl.scala#L404=) where the `sendOnboardingMail` is used an maybe remap the exception he throws with a custom one (like `sendOnboardingMail(destinationMails, pdf, onboardingMailParameters).leftMap(_ => new CustomException)`) and handle it specifically in the `onComplete` callbacks registered in the various endpoints that use the `performOnboardingWithSignature` function.

Reply:
as I said, logging the exception is the only option atm